### PR TITLE
changed actor lifespan,responsetime to use Timer instead of Histogram

### PR DESCRIPTION
**Pull request by vanjoeyg**

### CODE REVIEW

File: src/main/java/cloud/orbit/actors/extensions/metrics/dropwizard/OrbitActorExtension.java [hunk 1]
```diff
@@ -48,25 +49,26 @@
...
 
     //Actor Counter, per actor type and node, actor count breakdown to each type
     Map<String, Counter> actorCounters = new HashMap<>();
 
-    //Timer, per actor instance
-    Map<String, Long> actorStartTimes = new ConcurrentHashMap<>();
-
-    Map<String, Histogram> actorLifeSpanHistograms = new HashMap<>();
+    //Timer context, per actor instance
+    Map<String, Long> actorLifespanStartTimes = new ConcurrentHashMap<>();
```
Comment (ShawnTuatara) [inline src/main/java/cloud/orbit/actors/extensions/metrics/dropwizard/OrbitActorExtension.java]: We don't need to keep reference to the Timer object. We create a Timer.Context by calling the Timer.time(). Then when the actor deactivates we call the Timer.Context.stop().

http://metrics.dropwizard.io/3.1.0/getting-started/#timers

Comment (ShawnTuatara) [reply src/main/java/cloud/orbit/actors/extensions/metrics/dropwizard/OrbitActorExtension.java]: Ah... I see what is confusing me... You are actually not keeping a Timer for each actor. You are keeping a Timer for the actor "type" which in our case is the DefaultAccountActor and the timer is for that actor type. Lets re-word the field to better represent what we are keeping track of in the collection.

Comment (vanjoeyg) [reply src/main/java/cloud/orbit/actors/extensions/metrics/dropwizard/OrbitActorExtension.java]: 1. you can do Timer.Context.time()/stop(), then we have to keep the reference for the contexts plus it only does timer in nano-seconds
2. changed the variable name and added some comments

**Comment by ShawnTuatara** [comment]
https://github.com/orbit/orbit-dropwizard-metrics/pull/18

Created a separate pull request as I made changes to reflect how I was expecting to use Timer.Context and then added an activation timer.

File: src/main/java/cloud/orbit/actors/extensions/metrics/dropwizard/OrbitMetricsInvocationHandler.java [hunk 2]
```diff
@@ -37,53 +38,60 @@
...
-            actorMethodResponseTimeHistograms.put(histKey, hist);
+        String timerKey = getActorMethodResponseTimeMetricKey(actorClass, method.getName());
+        Timer timer = actorMethodResponseTimeTimers.get(timerKey);
+        if (null == timer)
+        {
+            timer = MetricsManager.getInstance().getRegistry().timer(timerKey);
+            actorMethodResponseTimeTimers.put(timerKey, timer);
         }
-        hist.update(durationMs.intValue());
+        timer.update(durationMs.longValue(), TimeUnit.MILLISECONDS);
```
Comment (ShawnTuatara) [inline src/main/java/cloud/orbit/actors/extensions/metrics/dropwizard/OrbitMetricsInvocationHandler.java:66]: Shouldn't need to convert to millis just to send it to a method that will most likely convert it back to nanos as that is how the timer internally represents the time.

**Comment by vanjoeyg** [comment]
close it and will create another one

**Pull request closed without merging**